src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -258,9 +258,9 @@ internal WatermarkOffsets GetWatermarkOffsets(string topic, int partition)
             return new WatermarkOffsets(low, high);
         }
 
-        internal void Subscribe(ICollection<string> topics)
+        internal void Subscribe(IEnumerable<string> topics)
         {
-            IntPtr list = LibRdKafka.topic_partition_list_new((IntPtr) topics.Count);
+            IntPtr list = LibRdKafka.topic_partition_list_new((IntPtr) topics.Count());
```

Comment:
`Count()` is less efficient in general of course, though after a quick scan through I don't see anywhere this is going to be of practical concern. This PR is LGTMing, but i'm going to hold off merging 'till I have a bit more time to review carefully.